plat/nvidia/tegra/common/drivers/smmu/smmu.c

```diff
@@ -449,11 +106,12 @@ void tegra_smmu_save_context(uint64_t smmu_ctx_addr)
 
 	assert((smmu_ctx_addr >= tzdram_base) && (smmu_ctx_addr <= tzdram_end));
 
-	/* index of _END_OF_TABLE_ */
-	smmu_ctx_regs[0].val = ARRAY_SIZE(smmu_ctx_regs) - 1;
+	/* get SMMU context table */
+	smmu_ctx_regs = plat_get_smmu_ctx();
+	assert(smmu_ctx_regs);
 
 	/* save SMMU register values */
-	for (i = 1; i < ARRAY_SIZE(smmu_ctx_regs) - 1; i++)
+	for (i = 1; i < smmu_ctx_regs[0].val; i++)
 		smmu_ctx_regs[i].val = mmio_read_32(smmu_ctx_regs[i].reg);
 
 	/* Save SMMU config settings */
```

Comment:
@vwadekar 
Cannot explain the timing, but our static analysis has raised an issue with this change:

> CID 143577 (#1 of 1): Sizeof not portable (SIZEOF_MISMATCH)suspicious_sizeof: Passing argument smmu_ctx_regs of type smmu_regs_t * and argument 8UL /* sizeof (smmu_ctx_regs) */ to function memcpy16 is suspicious. In this case, sizeof (smmu_regs_t *) is equal to sizeof (smmu_regs_t), but this is not a portable assumption.
>    Did you intend to use sizeof (*smmu_ctx_regs) instead of sizeof (smmu_ctx_regs)?
`        memcpy16((void *)(uintptr_t)smmu_ctx_addr, (void *)smmu_ctx_regs,
                 sizeof(smmu_ctx_regs));`

Can you confirm the intended use of sizeof here, lines 118/119?